corehq/apps/email/views.py

```diff
@@ -49,7 +52,7 @@ def post(self, request, *args, **kwargs):
         if form.is_valid():
             form.instance.domain = self.domain
             if form.instance.use_tracking_headers and not form.instance.sns_secret:
-                form.instance.sns_secret = "secret"
+                form.instance.sns_secret = token_urlsafe(16)
```

Comment:
Is this going to regenerate a new token every-time the form is saved?

Reply:
No, it will be generated once. Next time when form is submitted, secret will be there from the pre-filled data.

Reply:
oh right, because it checks `form.instance.sns_secret`, Cool thanks!